test/ngMock/angular-mocksSpec.js

```diff
@@ -2408,9 +2408,134 @@ describe('ngMockE2E', function() {
   });
 });
 
+
 describe('make sure that we can create an injector outside of tests', function() {
   //since some libraries create custom injectors outside of tests,
   //we want to make sure that this is not breaking the internals of
   //how we manage annotated function cleanup during tests. See #10967
   angular.injector([function($injector) {}]);
 });
+
+
+describe('don\'t leak memory between tests', function() {
+  var jQuery = window.jQuery;
+  var prevRootElement;
+  var prevRemoveDataSpy;
+  var prevCleanDataSpy;
+
+  it('should run a test to keep track of `removeData()`/`cleanData()` calls for later inspection',
+    function() {
+      module(function($provide) {
+        $provide.decorator('$rootElement', function($delegate) {
+          // Spy on `.removeData()` and `jQuery.cleanData()`,
+          // so the next test can verify that they have been called as necessary
+          prevRootElement = $delegate;
+          prevRemoveDataSpy = spyOn($delegate, 'removeData').andCallThrough();
+
+          if (jQuery) {
+            prevCleanDataSpy = spyOn(jQuery, 'cleanData').andCallThrough();
+          }
+
+          return $delegate;
+        });
+      });
+
+      // Inject the `$rootElement` to ensure it has been created
+      inject(function($rootElement) {
+        expect($rootElement.injector()).toBeDefined();
+      });
+    }
+  );
+
+  it('should clean up `$rootElement`-related data after each test', function() {
+    // One call is made by `testabilityPatch`'s `dealoc()`
+    // We want to verify the subsequent call, made by `angular-mocks`
+    // (Since `testabilityPatch` re-wrapps `$rootElement` and because jQuery
+    //  returns a different object, we don't capture the 1st call when using jQuery)
+    expect(prevRemoveDataSpy.callCount).toBe(jQuery ? 1 : 2);
+
+    if (jQuery) {
+      // One call is made by `testabilityPatch`'s `dealoc()`
+      // We want to verify the subsequent call, made by `angular-mocks`
+      expect(prevCleanDataSpy.callCount).toBe(2);
+
+      var cleanUpElems = prevCleanDataSpy.calls[1].args[0];
+      expect(cleanUpElems.length).toBe(1);
+      expect(cleanUpElems[0][0]).toBe(prevRootElement[0]);
+    }
+  });
```

Comment:
Theoretically, no. Jasmine will restore the original objects after each test. We hold on to the spies, but the original object is restored. When the `define` blocks are done, the references to those spies are useless (and should be garbage collected), but don't affect the environment in any way.